File: src/main/java/gg/grounds/domain/model/ProxyHeartbeatEntity.java
package gg.grounds.domain.model;

import io.quarkus.hibernate.reactive.panache.PanacheEntityBase;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "proxy_heartbeats", schema = "status")
public class ProxyHeartbeatEntity extends PanacheEntityBase {

    @Id
    public String proxyId;

    public int playerCount;
    public boolean healthy;
    public String version;

    @Column(columnDefinition = "jsonb")
    public String metaJson;

    public Instant lastSeenAt;
}

File: src/main/kotlin/gg/grounds/domain/model/ProxyHeartbeatEntity.kt
package gg.grounds.domain.model

import io.quarkus.hibernate.reactive.panache.PanacheEntityBase
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.time.Instant

@Entity
@Table(name = "proxy_heartbeats", schema = "status")
class ProxyHeartbeatEntity : PanacheEntityBase() {
    @Id
    @Column(name = "proxy_id")
    var proxyId: String? = null

    @Column(name = "player_count")
    var playerCount: Int = 0
    var healthy: Boolean = false
    var version: String? = null

    @Column(name = "meta", columnDefinition = "jsonb")
    var metaJson: String? = null

    @Column(name = "last_seen_at")
    var lastSeenAt: Instant? = null
}
